subworkflows/local/busco_diamond_blastp.nf

```diff
@@ -1,157 +1,94 @@
-    tbl_e = dir.filter { "$it" =~ /eukaryota_odb10/ }.map { meta,e,f -> [meta, "$e/$f/run_eukaryota_odb10/full_table.tsv", "$e/$f/run_eukaryota_odb10/eukaryota_odb10_full_table.tsv"] }.collect()
-    // create copies of full tables with a lineage identifier, avoids file name collision
-    tbl_a = tbl_a.map { meta,t,u -> [meta,file("$t").copyTo("$u")] }.collect()
-    tbl_b = tbl_b.map { meta,t,u -> [meta,file("$t").copyTo("$u")] }.collect()
-    tbl_e = tbl_e.map { meta,t,u -> [meta,file("$t").copyTo("$u")] }.collect()
-    // combine all three channels into a single channel: tuple( meta, a, b, e )
-    tbl_ab = tbl_a.combine(tbl_b, by:0)
-    tbl_abe = tbl_ab.combine(tbl_e, by:0)
-
-    // filter paths to busco_sequences for archaea, bacteria and eukaryota
-    seq_a = dir.filter { "$it" =~ /archaea_odb10/ }.map { meta,a,f -> [meta, "$a/$f/run_archaea_odb10/busco_sequences", "$a/$f/run_archaea_odb10/archaea_odb10_busco_sequences"] }.collect()
-    seq_b = dir.filter { "$it" =~ /bacteria_odb10/ }.map { meta,b,f -> [meta, "$b/$f/run_bacteria_odb10/busco_sequences", "$b/$f/run_bacteria_odb10/bacteria_odb10_busco_sequences"] }.collect()
-    seq_e = dir.filter { "$it" =~ /eukaryota_odb10/ }.map { meta,e,f -> [meta, "$e/$f/run_eukaryota_odb10/busco_sequences", "$e/$f/run_eukaryota_odb10/eukaryota_odb10_busco_sequences"] }.collect()
-    // create copies of busco_sequences with a lineage identifier, avoids file name collision
-    seq_a = seq_a.map { meta,t,u -> [meta,file("$t").copyTo("$u")] }.collect()
-    seq_b = seq_b.map { meta,t,u -> [meta,file("$t").copyTo("$u")] }.collect()
-    seq_e = seq_e.map { meta,t,u -> [meta,file("$t").copyTo("$u")] }.collect()
-    // combine all three channels into a single channel: tuple( meta, a, b, e )
-    seq_ab = seq_a.combine(seq_b, by:0)
-    seq_abe = seq_ab.combine(seq_e, by:0)
-
-    // full_table.tsv of the first lineage from GOAT_TAXONSEARCH 
-    // get the first lineage 
-    first_lineage = ch_lineages_goat.map { meta,l -> [meta, l[0]] }
-    // channel: [meta, busco_dir, fasta_filename, first_lineage]
-    first_dir = dir.combine(first_lineage, by:0)
-    // get the path to the table and copy it with a new name, channel: [meta, path_to_first_table]
-    first_table = first_dir.filter { meta,d,f,l -> "$d" =~ /$l/ }.map { meta,d,f,l -> [meta,"$d/$f/run_${l}/full_table.tsv","$d/$f/run_${l}/${l}_full_table.tsv"] }.collect()
-    first_table = first_table.map { meta,t,u -> [meta,file("$t").copyTo("$u")] }.collect() 
-
-    // module: creates input paths for EXTRACT_BUSCO_GENES
-    TAR (
-    tbl_abe,
-    seq_abe
-    )
-
-    // module: extract busco genes
-    EXTRACT_BUSCO_GENES (
-    TAR.out.dir_abe
-    )
-    ch_versions = ch_versions.mix(EXTRACT_BUSCO_GENES.out.versions)
 
     //
-    // Runs diamond_blastp with the extracted busco genes
+    // Create input for BLOBTOOLKIT_EXTRACTBUSCOS
     //
+    TARGZ ( BUSCO.out.seq_dir )
```

Comment:
This currently creates shallow `.tar.gz` because the target (`busco_sequences`) is a symbolic link (that's the default way Nextflow stages the files in) which is stored as-is in the archive. There isn't anything else in the archive:
```
mm49@tol-head1:~/link/scratch123/nextflow/work/75/9500eae50700845f208ae626bef531 $ ls -l
total 12
lrwxrwxrwx 1 mm49 team328 194 May  1 17:47 busco_sequences -> /lustre/scratch123/tol/teams/tolit/users/mm49/nextflow/work/4d/76a82eeed3ea9e0e8f4e5f50d274b2/GCA_922984935.2-eutheria_odb10-busco/GCA_922984935.2.subset.fasta/run_eutheria_odb10/busco_sequences
-rw-r--r-- 1 mm49 team328 308 May  1 17:47 busco_sequences.tar.gz
-rw-r--r-- 1 mm49 team328  68 May  1 17:47 versions.yml

mm49@tol-head1:~/link/scratch123/nextflow/work/75/9500eae50700845f208ae626bef531 $ tar tvzf busco_sequences.tar.gz 
lrwxrwxrwx mm49/team328      0 2023-05-01 17:47 busco_sequences -> /lustre/scratch123/tol/teams/tolit/users/mm49/nextflow/work/4d/76a82eeed3ea9e0e8f4e5f50d274b2/GCA_922984935.2-eutheria_odb10-busco/GCA_922984935.2.subset.fasta/run_eutheria_odb10/busco_sequences
```

I can imagine a few workarounds:

1. Pass de `-h` option to `tar`. In theory that would apply to symbolic links inside the directory structure too, but there aren't any in this case.
2. In the module, archive `$input/*` rather than `$input`. In theory that would miss hidden files in `$input`, but there aren't any in this case.
3. Modify the staging method to do a copy, though that will make the module very I/O inefficient (potentially thousands of files to copy)

Reply:
By the way, this probably explains why extract_busco_genes can't find anything. @alxndrdiaz had the same issue at some point

Reply:
> By the way, this probably explains why extract_busco_genes can't find anything. @alxndrdiaz had the same issue at some point

@muffato @priyanka-surana. the[ following code ](https://github.com/sanger-tol/blobtoolkit/blob/view/modules/local/tar.nf) is the one that fixed the issue  (`seq` is the path to the files that will be compressed, `fname="busco_sequences"`, and  `parent` is only the name of the lineage: `eukaryota_odb10`, etc):

```
# create a local copy of the directory, (RL  Recursive+always follow symbolic links in SOURCE )
cp -RL $seq $fname
# compress 
tar czf "$fname".tar.gz $fname
# move compressed file to the expected lineage folder
cp -RL "$fname".tar.gz $parent/"$fname".tar.gz
# remove local copy
rm -rf "$fname"*
```
This allows to compress the actual files instead of the symbolic link, but `tar -h` might work as well.  This will be very slow when copying all the sequences from all busco lineages.

Reply:
I don't think the `test` profile even with a working `tar` gets any hits. So will try the different options with the `test_full` profile. @alxndrdiaz let me know if this is not the case.

Reply:
The `test` profile should at least run diamond. Right now it is skipped because the input Fasta file is empty.

Also, I just noticed that Busco is being run with the `--tar` option, which means that the `busco_sequences` directories contain `fragmented_busco_sequences.tar.gz`,  `multi_copy_busco_sequences.tar.gz`, and  `single_copy_busco_sequences.tar.gz` instead of the respective directories with Fasta files. This is not compatible with the Python script extract-busco-genes, which expects all the Fasta files from all three to be directly available in its tar archive.

Reply:
Since these 3 tar archives are empty, I expect the resulting fasta file to be empty which would mean that diamond does not run. 

I will run the busco without the `tar` option, but will also skip publishing these to results because without the `tar` option the busco output has too many files.

Reply:
I removed the `-tar` option in `busco`  + for `tar` module added both `-h` and `input/*` options. And now the `DIAMOND_BLASTP` does start but ends in error.
```
Error: Error reading input stream at line 8: FASTA format error: Missing '>' at record start.
```
But the fasta file (`/lustre/scratch123/tol/teams/tolit/users/ps22/pipelines/testing/work/55/92997681eb24f40ba5b2b9a85e6eb5/GCA_927399515.1_buscogenes.fasta`) has the correct format – no missing ">". 

I am going to remove `input/*` and see if `DIAMOND_BLASTP` still gets started. Trying to do minimal changes. Then will work on the `DIAMOND` issues 🤔

Reply:
> I don't think the `test` profile even with a working `tar` gets any hits. So will try the different options with the `test_full` profile. @alxndrdiaz let me know if this is not the case.

There should be at least one hit as far as I remember.

Reply:
Yes I am getting that now. Thanks 👍🏼